src/analyze/analyze.c

```diff
@@ -1213,7 +1239,99 @@ static int dot(int argc, char *argv[], void *userdata) {
+                { "Color", "Triggers",          config_parse_string,  0, &arg_dependency_color[UNIT_TRIGGERS]  },
+                { "Style", "After",             config_parse_string,  0, &arg_dependency_style[UNIT_AFTER]     },
+                { "Style", "Requires",          config_parse_string,  0, &arg_dependency_style[UNIT_REQUIRES]  },
+                { "Style", "Requisite",         config_parse_string,  0, &arg_dependency_style[UNIT_REQUISITE] },
+                { "Style", "Wants",             config_parse_string,  0, &arg_dependency_style[UNIT_WANTS]     },
+                { "Style", "Conflicts",         config_parse_string,  0, &arg_dependency_style[UNIT_CONFLICTS] },
+                { "Style", "BindsTo",           config_parse_string,  0, &arg_dependency_style[UNIT_BINDS_TO]  },
+                { "Style", "PartOf",            config_parse_string,  0, &arg_dependency_style[UNIT_PART_OF]   },
+                { "Style", "OnFailure",         config_parse_string,  0, &arg_dependency_style[UNIT_ON_FAILURE]},
+                { "Style", "Triggers",          config_parse_string,  0, &arg_dependency_style[UNIT_TRIGGERS]  },
+                {}
+        };
+
+        static const char* colors[] = {
+                "black",    /* UNIT_REQUIRES */
+                "blue",     /* UNIT_REQUISITE */
+                "grey66",   /* UNIT_WANTS */
+                "magenta",  /* UNIT_BINDS_TO */
+                "black",    /* UNIT_PART_OF */
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                "red",      /* UNIT_CONFLICTS */
+                NULL,
+                NULL,
+                "green",    /* UNIT_AFTER */
+                "orange",   /* UNIT_ON_FAILURE */
+                "grey66"    /* UNIT_TRIGGERS */
+        };
+
+        for (UnitDependency dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                if (colors[dependency]) {
+                        arg_dependency_color[dependency] = strdup(colors[dependency]);
+
+                        if (dependency == UNIT_PART_OF)
+                                arg_dependency_style[dependency] = strdup("dashed");
+                        else if (dependency == UNIT_TRIGGERS)
+                                arg_dependency_style[dependency] = strdup("dotted");
+                        else
+                                arg_dependency_style[dependency] = strdup("solid");
+
+                        if (!arg_dependency_color[dependency] || !arg_dependency_style[dependency]) {
+                            return log_oom();
+                        }
+                }
+        }
+
+        /* colors and styles may be overridden on cmd line */
+        STRV_FOREACH(arg_dot_dep_color_style, arg_dot_dependency_colors_styles) {
+                char *ptr = strtok(*arg_dot_dep_color_style, delim);
```

Comment:
this is not ok. strtok() is not ok in our codebase, see below.

Also, please parse this when parsing the cmdline. Don't wait until you actually need it. also, why have arg_dot_dependency_colors_styles at all? why isn't arg_dependency_color[] and arg_dependency_style[] appropriate for that?

Reply:
If these should be parsed when parsing the cmdline, i.e. in parse_argv(), and arg_dependency_color[] / arg_dependency_style[] should be used instead of arg_dot_dependency_colors_styles, then the defaults already must have been loaded, so the given cmdline arguments can override them.
Defaults are currently loaded in dot() and would need to be done in run() instead what doesn't make much sense if not running 'dot'.

Reply:
now parsing when parsing the cmdline.